### DISCUSSION TITLE
getNextEvents returns xstate.done.event

### DISCUSSION THREAD

**Discussion by stovenator** [statelyai/xstate#5133] [Q&A]
Hello,

I have a state machine with a parallel event that has an onDone event to trigger when all of the parallel tasks reach their final state. When I use the getNextEvents function from [the migration page](https://stately.ai/docs/migration#statenextevents-has-been-removed), it returns the available events, as well as an xstate.done.event . 

Example using the Coffee Machine:

```
import { createMachine } from 'xstate';

export const machine = createMachine({
  id: 'coffee',
  initial: 'preparing',
  states: {
    preparing: {
      states: {
        grindBeans: {
          initial: 'grindingBeans',
          states: {
            grindingBeans: {
              on: {
                BEANS_GROUND: {
                  target: 'beansGround',
                },
              },
            },
            beansGround: {
              type: 'final',
            },
          },
        },
        boilWater: {
          initial: 'boilingWater',
          states: {
            boilingWater: {
              on: {
                WATER_BOILED: {
                  target: 'waterBoiled',
                },
              },
            },
            waterBoiled: {
              type: 'final',
            },
          },
        },
      },
      type: 'parallel',
      onDone: {
        target: 'makingCoffee',
      },
    },
    makingCoffee: {},
  },
});
```

When I call getNextEvents() the return is.

```
{
  "next": [
    "xstate.done.state.coffee.preparing",
    "BEANS_GROUND",
    "WATER_BOILED"
  ]
}
```

The  "xstate.done.state.coffee.preparing" event is not actually an event that can be sent to the actor.

Is there a reason the onDone event is listed here? Shouldn't xstate filter these xstate.done.event items from the list?

**Comment by davidkpiano** [comment]
They should be filtered. A workaround is to filter events that don't start with `xstate.`
